File: android/app/src/main/java/com/schoolfinder/app/data/SessionStore.java
package com.schoolfinder.app.data;

import android.content.Context;
import android.content.SharedPreferences;

/** Persists the dev-auth session in SharedPreferences and mirrors it into SessionHolder. */
public class SessionStore {

    private final SharedPreferences prefs;

    public SessionStore(Context context) {
        prefs = context.getApplicationContext().getSharedPreferences("session", Context.MODE_PRIVATE);
    }

    /** Loads any saved session into SessionHolder; returns it (or null). */
    public Session load() {
        String uid = prefs.getString("uid", null);
        if (uid == null) return null;
        Session session = new Session(
                uid,
                prefs.getString("name", uid),
                prefs.getString("role", "STUDENT"));
        SessionHolder.current = session;
        return session;
    }

    public void save(Session session) {
        prefs.edit()
                .putString("uid", session.uid)
                .putString("name", session.displayName)
                .putString("role", session.role)
                .apply();
        SessionHolder.current = session;
    }

    public void clear() {
        prefs.edit().clear().apply();
        SessionHolder.current = null;
    }
}

File: android/app/src/main/java/com/schoolfinder/app/data/SessionStore.kt
package com.schoolfinder.app.data

import android.content.Context
import androidx.datastore.preferences.core.edit
import androidx.datastore.preferences.core.stringPreferencesKey
import androidx.datastore.preferences.preferencesDataStore
import kotlinx.coroutines.flow.first

private val Context.dataStore by preferencesDataStore(name = "session")

/** Persists the dev-auth session and mirrors it into [SessionHolder] for the network layer. */
class SessionStore(private val context: Context) {

    private val uidKey = stringPreferencesKey("uid")
    private val nameKey = stringPreferencesKey("name")
    private val roleKey = stringPreferencesKey("role")

    /** Loads any saved session into [SessionHolder]; returns it (or null). */
    suspend fun load(): Session? {
        val prefs = context.dataStore.data.first()
        val uid = prefs[uidKey] ?: return null
        val session = Session(
            uid = uid,
            displayName = prefs[nameKey] ?: uid,
            role = prefs[roleKey] ?: "STUDENT",
        )
        SessionHolder.current = session
        return session
    }

    suspend fun save(session: Session) {
        context.dataStore.edit {
            it[uidKey] = session.uid
            it[nameKey] = session.displayName
            it[roleKey] = session.role
        }
        SessionHolder.current = session
    }

    suspend fun clear() {
        context.dataStore.edit { it.clear() }
        SessionHolder.current = null
    }
}
